### PR TITLE
Draw less perform more

### PR DESCRIPTION
**Pull request by pburnsdata**
# Issues
#99
#100 
#104

### CODE REVIEW

File: src/onDraw/addRowDisplayToggle.js [hunk 1]
```diff
@@ -25,17 +29,15 @@ export default function addRowDisplayToggle() {
             .map(d);
     }
 
-    var childNest = iterateNest(chart.data.raw, 0);
+    var childNest = iterateNest(chart.data.summarized, 0);
```
Comment (samussiah) [inline src/onDraw/addRowDisplayToggle.js]: Generally speaking, throw a comment that describes each data structure, how it's nested, whether it's an object or an array, etc.  There's just so much going on with the expandable rows that I have to figure out what's happening at each step. /whining

File: src/onDraw/addRowDisplayToggle.js [hunk 2]
```diff
@@ -46,6 +48,7 @@ export default function addRowDisplayToggle() {
         d.id.split('  |').forEach(function(level) {
             currentNest = currentNest[level];
```
Comment (samussiah) [inline src/onDraw/addRowDisplayToggle.js]: clever

File: src/onDraw/addRowDisplayToggle.js [hunk 3]
```diff
@@ -58,13 +61,68 @@ export default function addRowDisplayToggle() {
...
+            var childrenData = chart.data.summarized.filter(
+                a => childIds.includes(a.id) && (a.filtered != true || a.visible_child)
+            );
+
+            row.classed('selected', true);
+
+            // repeating *s to place children after their parent in the correct order
+            childrenData.forEach((childData, i) =>
+                chart.tbody
+                    .insert('tr', '.selected' + ' + *'.repeat(i + 1))
```
Comment (samussiah) [inline src/onDraw/addRowDisplayToggle.js]: also clever

Comment (samussiah) [reply src/onDraw/addRowDisplayToggle.js]: never heard of the `+` selector, cool stuff

**Comment by samussiah** [comment]
Great work @pburnsdata !  Found a few bugs:

- When nesting by folder and form, hitting Expand All places all forms after all folders.
- Sometime Expand All reverts to a previous nesting when it's un-checked.

Also made some suggestions for modularizing `addRowDisplayToggle` and throwing in comments describing the data structure every step of the way.

File: src/onDraw/addRowDisplayToggle.js [hunk 4]
```diff
@@ -1,3 +1,7 @@
+import customizeRows from './customizeRows';
+import customizeCells from './customizeCells';
+import flagParentRows from './flagParentRows';
+
 export default function addRowDisplayToggle() {
     var chart = this;
     var config = this.config;
```
[the remark this answers is not in the document]
Comment (pburnsdata) [reply src/onDraw/addRowDisplayToggle.js]: good idea

File: src/onDraw/addRowDisplayToggle.js (hunk 1 above)
Comment (pburnsdata) [reply src/onDraw/addRowDisplayToggle.js]: Makes perfect sense - ill add that in

**Comment by samussiah** [comment]
per discussion with @pburnsdata , we decided to optionalize the visit sort order variable because nested visit rows behave unusually without that variable

**Pull request merged**